hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -573,11 +688,25 @@ int readRemote(long position, byte[] b, int offset, int length, TracingContext t
     } catch (AzureBlobFileSystemException ex) {
       if (ex instanceof AbfsRestOperationException) {
         AbfsRestOperationException ere = (AbfsRestOperationException) ex;
+        abfsHttpOperation = ((AbfsRestOperationException) ex).getAbfsHttpOperation();
         if (ere.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
           throw new FileNotFoundException(ere.getMessage());
         }
+        /*
+         * Status 416 is sent when read range is out of contentLength range.
+         * This would happen only in the case if contentLength is not known before
+         * opening the inputStream.
+         */
+        if (ere.getStatusCode() == READ_PATH_REQUEST_NOT_SATISFIABLE
+            && !getFileStatusInformationPresent()) {
+          return -1;
+        }
       }
       throw new IOException(ex);
+    } finally {
+      if (!getFileStatusInformationPresent() && abfsHttpOperation != null) {
```

Comment:
can you move condition` !getFileStatusInformationPresent()` to inside `initPropertiesFromReadResponseHeader`

```
void initPropertiesFromReadResponseHeader(abfsHttpOperation) {
   if(!getFileStatusInformationPresent()) {
       return;
   }
//...
}
```

Reply:
Would suggest if still keep the hasFileStatusInfo check outside, so the reader of code understands that this piece of code should only be called if fileStatus is not already there.

Open to put in the method. What you feel?

Thanks!